Content.Server/SimpleStation14/Species/Shadowkin/Systems/ShadowkinSystem.cs

```diff
@@ -1,157 +1,205 @@
+
+    public override void Initialize()
     {
-        [Dependency] private readonly ShadowkinPowerSystem _powerSystem = default!;
-        [Dependency] private readonly IRobustRandom _random = default!;
-        [Dependency] private readonly IEntityManager _entityManager = default!;
+        base.Initialize();
 
-        public override void Initialize()
-        {
-            base.Initialize();
+        SubscribeLocalEvent<ShadowkinComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<ShadowkinComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<ShadowkinComponent, ComponentShutdown>(OnShutdown);
+    }
 
-            SubscribeLocalEvent<ShadowkinComponent, ExaminedEvent>(OnExamine);
-            SubscribeLocalEvent<ShadowkinComponent, ComponentInit>(OnInit);
+    private void OnExamine(EntityUid uid, ShadowkinComponent component, ExaminedEvent args)
+    {
+        if (!args.IsInDetailsRange)
+            return;
+
+        var powerType = ShadowkinPowerSystem.GetLevelName(component.PowerLevel);
+
+        if (args.Examined == args.Examiner)
+        {
+            args.PushMarkup(Loc.GetString("shadowkin-power-examined-self",
+                ("power", (int) component.PowerLevel),
+                ("powerMax", component.PowerLevelMax),
+                ("powerType", powerType)
+            ));
         }
+        else
+        {
+            args.PushMarkup(Loc.GetString("shadowkin-power-examined-other",
+                ("target", Identity.Entity(uid, _entity)),
+                ("powerType", powerType)
+            ));
+        }
+    }
+
+    private void OnInit(EntityUid uid, ShadowkinComponent component, ComponentInit args)
+    {
+        if (component.PowerLevel <= ShadowkinComponent.PowerThresholds[ShadowkinPowerThreshold.Min] + 1f)
+            _power.SetPowerLevel(uid, ShadowkinComponent.PowerThresholds[ShadowkinPowerThreshold.Okay]);
+
+        component.MaxedPowerAccumulator = _random.NextFloat(component.MaxedPowerRateMin, component.MaxedPowerRateMax);
+        component.MinPowerAccumulator = _random.NextFloat(component.MinRateMin, component.MinRateMax);
+    }
 
-        private void OnExamine(EntityUid uid, ShadowkinComponent component, ExaminedEvent args)
+    private void OnShutdown(EntityUid uid, ShadowkinComponent component, ComponentShutdown args)
+    {
+        _power.UpdateAlert(component.Owner, false);
+    }
+
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = _entity.EntityQuery<ShadowkinComponent>();
+
+        // Update power level for all shadowkin
+        foreach (var component in query)
         {
-            if (args.IsInDetailsRange)
+            var oldPowerLevel = ShadowkinPowerSystem.GetLevelName(component.PowerLevel);
+
+            if (!component.Blackeye)
+                _power.TryBlackeye(component.Owner);
+            _power.TryUpdatePowerLevel(component.Owner, frameTime);
```

Comment:
Raise an event upon switching from one level of power to another. Subscribe to that event with a Blackeye system. If the event is to the lowest amount of charge (and after checking the actual charge if required), black eye.

Do not do this.